test/datasets/test_dbpedia.py

```diff
@@ -0,0 +1,85 @@
+import os
+import random
+import string
+import tarfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.dbpedia import DBpedia
+
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "DBpedia")
+    temp_dataset_dir = os.path.join(base_dir, "dbpedia_csv")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ["train.csv", "test.csv"]:
+        csv_file = os.path.join(temp_dataset_dir, file_name)
+        mocked_lines = mocked_data[os.path.splitext(file_name)[0]]
+        with open(csv_file, "w") as f:
+            for i in range(5):
+                label = seed % 11
+                rand_string = " ".join(
+                    random.choice(string.ascii_letters) for i in range(seed)
+                )
+                dataset_line = (label, f"{rand_string}")
+                f.write(f'{label},"{rand_string}"\n')
+
+                # append line to correct dataset split
+                mocked_lines.append(dataset_line)
+                seed += 1
+
+    compressed_dataset_path = os.path.join(base_dir, "dbpedia_csv.tar.gz")
+    # create gz file from dataset folder
+    with tarfile.open(compressed_dataset_path, "w:gz") as tar:
```

Comment:
We can actually do something similar to what we're doing for the [AmazonReviewPolarity test ](https://github.com/Nayef211/text/blob/main/test/datasets/test_amazonreviewpolarity.py#L42-L44). This way we don't need the additional for-loop

Reply:
That's better, thanks